dephell_archive/_stream.py

```diff
@@ -14,15 +15,35 @@ class ArchiveStream:
     mode = attr.ib(type=str, default='r')
     encoding = attr.ib(type=Optional[str], default=None)
 
-    def exists(self) -> bool:
-        try:
+    def _get_info(self):
+        with suppress(KeyError):
             if hasattr(self.descriptor, 'getmember'):
-                self.descriptor.getmember(str(self.member_path))  # tar
-            else:
-                self.descriptor.getinfo(str(self.member_path))  # zip
-        except KeyError:
+                return self.descriptor.getmember(str(self.member_path))  # tar
+            return self.descriptor.getinfo(str(self.member_path))  # zip
+        return None
+
+    def exists(self) -> bool:
+        return self._get_info() is not None
+
+    def is_file(self) -> bool:
+        info = self._get_info()
+        if info is None:
+            return False
+        # tar
+        if hasattr(info, 'isfile'):
+            return info.isfile()
+        # zip
+        return info.filename[-1] != '/'
+
+    def is_dir(self) -> bool:
```

Comment:
this fails for `graphviz-0.13.2`

Reply:
Thank you for the review! Let's take care of it in #8 since it depends on how the stream was created, and it affects the code in #6.

Reply:
I've split it off to a new PR.